### PR TITLE
Speedy details

### PR DESCRIPTION
**Pull request by tvrg**
image

### CODE REVIEW

File: database/test.go [hunk 1]
```diff
@@ -80,22 +80,35 @@ func InsertTests(db *sqlx.DB, tests []model.TestResult, upload model.Upload) err
 	return nil
 }
 
+func UpdateFlakyTests(db *sqlx.DB, commit model.Commit) error {
```
Comment (jmattheis) [inline database/test.go]: Maybe only pass the commit.id into this method?

File: api/report.go [hunk 2]
```diff
@@ -80,5 +80,11 @@ func (a *api) AddReport(ctx echo.Context, commitSha string, params AddReportPara
 			fmt.Sprintf("couldn't store test results: %s", err))
 	}
 
+	err = database.UpdateFlakyTests(a.db, *commit)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest,
```
Comment (jmattheis) [inline api/report.go]: Shouldn't this be an internal server error?

File: api/api.go [hunk 3]
```diff
@@ -51,15 +52,28 @@ func New(db *sqlx.DB, token string) *echo.Echo {
 			"Flakes": flakes,
 		})
 	})
-	app.GET("/ui/test/:name/fails", func(c echo.Context) error {
-		name := c.Param("name")
-		fails, err := api.failedBuilds(name)
+	app.GET("/ui/test/:id/fails", func(c echo.Context) error {
+		testID := c.Param("id")
+		fails, err := api.failedBuilds(testID)
```
Comment (jmattheis) [inline api/api.go]: fails looks like a weird name for this, maybe testFailure, or just test? test.ID, test.Name, test.Builds

File: openapi.yaml [hunk 4]
```diff
@@ -118,11 +151,30 @@ components:
         total_fails:
           type: integer
           description: The total number of failures
+    FailedBuilds:
```
Comment (jmattheis) [inline openapi.yaml]: Can we name this FailedTest or TestFailure?

**Review by jmattheis — APPROVED** [review]
:shipit:

File: api/report.go (hunk 2 above)
Comment (tvrg) [reply api/report.go]: True!

File: openapi.yaml (hunk 4 above)
Comment (tvrg) [reply openapi.yaml]: I've renamed FailedBuilds to TestResults and reused TestResult for FailedBuild.

File: api/api.go (hunk 3 above)
Comment (tvrg) [reply api/api.go]: True, renamed to test.

File: database/test.go (hunk 1 above)
Comment (tvrg) [reply database/test.go]: Yes!

**Pull request merged**